test/integration/scheduler/queue_test.go

```diff
@@ -587,26 +587,93 @@ func TestCoreResourceEnqueue(t *testing.T) {
+				// Create the node which will make it match pod1's podAffinity.
+				if _, err := testCtx.ClientSet.CoreV1().Nodes().Create(testCtx.Ctx, st.MakeNode().Name("fake-node2").Label("zone", "zone1").Obj(), metav1.CreateOptions{}); err != nil {
 					return nil, fmt.Errorf("failed to update pod1: %w", err)
 				}
-				return map[framework.ClusterEvent]uint64{framework.NodeLabelChange: 1}, nil
+				return map[framework.ClusterEvent]uint64{framework.NodeAdd: 1}, nil
 			},
 			wantRequeuedPods:          sets.New("pod1"),
 			enableSchedulingQueueHint: []bool{true},
 		},
+		{
+			name: "Pod rejected by the interPodAffinity plugin is requeued when updating the node with the topologyKey label of the pod affinity",
+			initialNodes: []*v1.Node{
+				st.MakeNode().Name("fake-node").Obj(),
+			},
+			initialPods: []*v1.Pod{
+				st.MakePod().Name("pod1").Label("affinity1", "affinity1").Container("image").Node("fake-node").Obj(),
+			},
+			pods: []*v1.Pod{
+				// - pod2, pod3 will be rejected by the PodAffinity plugin.
+				st.MakePod().Name("pod2").Label("affinity1", "affinity1").PodAffinityExists("affinity1", "node", st.PodAffinityWithRequiredReq).Container("image").Obj(),
+				st.MakePod().Name("pod3").Label("affinity1", "affinity1").PodAffinityExists("affinity1", "other", st.PodAffinityWithRequiredReq).Container("image").Obj(),
+			},
+			triggerFn: func(testCtx *testutils.TestContext) (map[framework.ClusterEvent]uint64, error) {
+				// Add "node" label to the node which may make pod2 schedulable.
+				if _, err := testCtx.ClientSet.CoreV1().Nodes().Update(testCtx.Ctx, st.MakeNode().Name("fake-node").Label("node", "fake-node").Obj(), metav1.UpdateOptions{}); err != nil {
+					return nil, fmt.Errorf("failed to update pod1: %w", err)
+				}
+				return map[framework.ClusterEvent]uint64{framework.NodeLabelChange: 1}, nil
+			},
+			wantRequeuedPods:          sets.New("pod2"),
+			enableSchedulingQueueHint: []bool{true},
+		},
+		{
+			name:         "Pod rejected by the interPodAffinity plugin is requeued when updating the node with the topologyKey label of the pod anti affinity",
+			initialNodes: []*v1.Node{st.MakeNode().Name("fake-node").Label("node", "fake-node").Label("service", "service-a").Label("other", "fake-node").Obj()},
+			initialPods: []*v1.Pod{
+				st.MakePod().Name("pod1").Label("anti1", "anti1").Container("image").Node("fake-node").Obj(),
+			},
+			pods: []*v1.Pod{
+				// - pod2, pod3, pod4, pod5 will be rejected by the PodAffinity plugin.
+				st.MakePod().Name("pod2").Label("anti1", "anti1").PodAntiAffinityExists("anti1", "node", st.PodAntiAffinityWithRequiredReq).Container("image").Obj(),
+				st.MakePod().Name("pod3").Label("anti1", "anti1").PodAntiAffinityExists("anti1", "service", st.PodAntiAffinityWithRequiredReq).Container("image").Obj(),
+				st.MakePod().Name("pod4").Label("anti1", "anti1").PodAntiAffinityExists("anti1", "region", st.PodAntiAffinityWithRequiredReq).Container("image").Obj(),
```

Comment:
https://github.com/kubernetes/kubernetes/pull/127470#discussion_r1800424000
I wanted to override this operation, but if there was no region label originally, pod4 should be scheduled directly and would not be rejected.

Reply:
Yeah, as @macsko also [pointed out](https://github.com/kubernetes/kubernetes/pull/127470#discussion_r1803103466), unmatch -> match scenario is invalid because unmatch node can accept the Pod in the first place.